Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -116,50 +112,54 @@ ROMol *deleteSubstructs(const ROMol &mol, const ROMol &query, bool onlyFrags,
 
   // if didn't find any matches nothing to be done here
   // simply return a copy of the molecule
-  if (fgpMatches.size() == 0) {
+  if (fgpMatches.empty()) {
     return res;
   }
 
-  for (mati = fgpMatches.begin(); mati != fgpMatches.end(); mati++) {
+  // all matches on the molecule - list of list of atom ids
+  VECT_INT_VECT matches;
+  matches.reserve(fgpMatches.size());
+  for (auto mati : fgpMatches) {
     INT_VECT match;  // each match onto the molecule - list of atoms ids
-    for (mi = mati->begin(); mi != mati->end(); mi++) {
-      match.push_back(mi->second);
+    match.reserve(mati.size());
+    for (auto mi : mati) {
+      match.push_back(mi.second);
     }
     matches.push_back(match);
   }
 
   // now loop over the list of matches and check if we can delete any of them
   INT_VECT delList;
-
-  VECT_INT_VECT_I mxi, fi;
   if (onlyFrags) {
     VECT_INT_VECT frags;
-
     MolOps::getMolFrags(*res, frags);
-    for (fi = frags.begin(); fi != frags.end(); fi++) {
-      std::sort(fi->begin(), fi->end());
-      for (mxi = matches.begin(); mxi != matches.end(); mxi++) {
-        std::sort(mxi->begin(), mxi->end());
-        if ((*fi) == (*mxi)) {
+    for (auto fi : frags) {
+      std::sort(fi.begin(), fi.end());
+      for (auto mxi : matches) {
+        std::sort(mxi.begin(), mxi.end());
+        if (fi == mxi) {
           INT_VECT tmp;
-          Union((*mxi), delList, tmp);
+          Union(mxi, delList, tmp);
           delList = tmp;
           break;
         }  // end of if we found a matching fragment
-      }    // endof loop over matches
+      }    // end of loop over matches
     }      // end of loop over fragments
-  }        // end of if onlyFrags
-  else {
+  } else {
     // in this case we want to delete any matches we find
     // simply loop over the matches and collect the atoms that need to
     // be removed
-    for (mxi = matches.begin(); mxi != matches.end(); mxi++) {
+    for (auto mxi : matches) {
       INT_VECT tmp;
-      Union((*mxi), delList, tmp);
+      Union(mxi, delList, tmp);
       delList = tmp;
     }
   }
 
+  if (delList.empty()) {
```

Comment:
Makes sense.

Should `commitBatchEdit()` _also_ early return if there is nothing added or deleted? Otherwise this seems like a bit of a footgun down the road.